client/wildcard/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useCallback } from 'react'
+import classNames from 'classnames'
+import ChevronLeftIcon from 'mdi-react/ChevronLeftIcon'
+import ChevronRightIcon from 'mdi-react/ChevronRightIcon'
+import useResizeObserver from 'use-resize-observer/polyfilled'
+import { useDebounce } from '@sourcegraph/wildcard'
+
+type Page = '...' | number
+
+interface PageButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+    active?: boolean
+}
+
+const PageButton: React.FunctionComponent<PageButton> = ({ children, active, ...props }) => {
+    const refocus = useCallback((button: HTMLButtonElement | null) => active && button?.focus(), [active])
+
+    return (
+        <li>
+            <button
+                type="button"
+                className={classNames('btn', 'mx-1', 'pagination__button', active && 'btn-primary')}
+                aria-current={active}
+                ref={refocus}
+                {...props}
+            >
+                {children}
+            </button>
+        </li>
+    )
+}
+
+const range = (start: number, end: number): number[] => {
+    const length = end - start + 1
+    return Array.from({ length }, (value, index) => start + index)
+}
+
+const getMiddlePages = ({ currentPage, maxPages, shrink }: GetPages): Page[] => {
+    const minimumSiblingRange = shrink ? 2 : 4
+    const maximumSiblingRange = shrink ? 3 : 5
+    const minimumBoundary = 2
+    const maximumBoundary = maxPages - 1
+
+    const siblingsStart = Math.max(
+        Math.min(
+            currentPage - minimumSiblingRange / 2,
+            maxPages - maximumSiblingRange // Extend range when page is high
+        ),
+        minimumBoundary
+    )
+
+    const siblingsEnd = Math.min(
+        Math.max(
+            currentPage + minimumSiblingRange / 2,
+            maximumSiblingRange + 1 // Extend range when page is low
+        ),
+        maximumBoundary
+    )
+
+    const middle: Page[] = range(siblingsStart, siblingsEnd)
+
+    if (middle[0] !== minimumBoundary) {
+        middle[0] = '...'
+    }
+
+    if (middle[middle.length - 1] !== maximumBoundary) {
+        middle[middle.length - 1] = '...'
+    }
+
+    return middle
+}
+
+interface GetPages extends Pick<Props, 'currentPage' | 'maxPages'> {
+    shrink: boolean
+}
+
+const getPages = ({ currentPage, maxPages, shrink }: GetPages): Page[] => {
+    const pages: Page[] = [1]
+
+    if (maxPages > 1) {
+        pages.push(maxPages)
+    }
+
+    if (maxPages > 2) {
+        pages.splice(1, 0, ...getMiddlePages({ currentPage, maxPages, shrink }))
+    }
+
+    return pages
+}
+
+export interface Props {
```

Comment:
```suggestion
export interface PaginationProps {
```

nit: since this is exported, I would recommend making a name that won't have to be aliased when importing.